python/ray/serve/_private/proxy.py

```diff
@@ -640,8 +640,17 @@ async def unary_unary(
                 else:
                     response = message
 
-            context.set_code(status.code)
-            context.set_details(status.message)
+            # Only the latest code and details will take effect. If the user already
+            # set them to a truthy value in the context, skip setting them with Serve's
+            # default values. By default, if nothing is set, the code is 0 and the
+            # details is "", which both are falsy. So if the user did not set them or
+            # if they're explicitly set to falsy values, such as None, Serve will
+            # continue to set them with our default values.
+            if not context.code():
+                context.set_code(status.code)
```

Comment:
One issue I see here: if the user sets a status code, but then their handler raises an exception after that, should the returned code be an internal error or the code they set?

I would expect that this would be an internal error, but not sure. Can you check what gRPC does in this case?

Reply:
Yes, this will raise internal error, will add a test case for it!